app/src/main/java/com/example/eyespeak/ActionButton.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.eyespeak
+
+import android.content.Context
+import android.speech.tts.TextToSpeech
+import android.util.AttributeSet
+import android.widget.Toast
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class ActionButton @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : EyeSpeakButton(context, attrs, defStyle) {
+
+    var action: (() -> Unit)? = null
+
+    init {
+        setOnClickListener {
+            Toast.makeText(context, text, Toast.LENGTH_LONG).show()
```

Comment:
We are duplicating logic in our custom button classes, we should factor that out eventually.

Reply:
We can address this in our cleanup ticket.